### PR TITLE
[ADAM-1352] Clean up consensus model usage.

### LINKED ISSUE #1352: ConsensusModel's can't really be instantiated
Their constructors are all package private. I think that's the right approach, but we should have a singleton class that creates them. I'll:

* Make a singleton to create consensus models
* Improve the known INDEL consensus model, because it needs a bit of TLC
* Add functions for merging multiple consensus models

I need this for https://github.com/bigdatagenomics/avocado/issues/200.

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #1352:

* Fixed issues with consensus model instantiation.
* Added union consensus model (to support multiple models).
* Cleaned up consensus model documentation.
* Call distinct in IndelRealigner on consensus sequences to eliminate dupes.
* Added test for known indel model. Fixed off by one bug.
* Add test for union model.

### CODE REVIEW

File: adam-cli/src/main/scala/org/bdgenomics/adam/cli/Transform.scala [hunk 1]
```diff
@@ -174,9 +174,9 @@ class Transform(protected val args: TransformArgs) extends BDGSparkCommand[Trans
 
       // fold and get the consensus model for this realignment run
       val consensusGenerator = Option(args.knownIndelsFile)
-        .fold(new ConsensusGeneratorFromReads().asInstanceOf[ConsensusGenerator])(
-          new ConsensusGeneratorFromKnowns(_, rdd.rdd.context).asInstanceOf[ConsensusGenerator]
-        )
+        .fold(ConsensusGenerator.fromReads)(file => {
+          ConsensusGenerator.fromKnownIndels(rdd.rdd.context.loadVariants(file))
+        })
```
Comment (heuermh) [inline adam-cli/src/main/scala/org/bdgenomics/adam/cli/Transform.scala]: this reads a lot better

File: adam-core/src/main/scala/org/bdgenomics/adam/models/IndelTable.scala [hunk 2]
```diff
@@ -76,11 +64,12 @@ private[adam] object IndelTable {
           val deletionLength = v.getReferenceAllele.length - v.getAlternateAllele.length
           val start = v.getStart + v.getAlternateAllele.length
 
-          Consensus("", ReferenceRegion(referenceName, start, start + deletionLength))
+          Consensus("", ReferenceRegion(referenceName, start, start + deletionLength + 1))
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/models/IndelTable.scala]: oops  :)  was this a bug, or something you found when cleaning up?

Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/models/IndelTable.scala]: This was a bug.

**Comment by fnothaft** [comment]
This is good to go now.

**Comment by heuermh** [comment]
Thank you, @fnothaft!

**Pull request merged**